src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -24,18 +25,18 @@
     private final String name;
     private final String phone;
     private final String email;
-    private final boolean isDone;
+    private final String isDone;
 
     /**
      * Constructs a {@code JsonAdaptedPerson} with the given person details.
      */
     @JsonCreator
     public JsonAdaptedPerson(@JsonProperty("name") String name, @JsonProperty("phone") String phone,
-                             @JsonProperty("email") String email, @JsonProperty("done") boolean isDone) {
+                             @JsonProperty("email") String email, @JsonProperty("isDone") String isDone) {
         this.name = name;
         this.phone = phone;
         this.email = email;
-        this.isDone = isDone;
+        this.isDone = isDone.toUpperCase(Locale.ROOT);
```

Comment:
Won't this cause a nullPointerException if isDone does not exist in the Json file?

Reply:
I think follow wat they did with the other fields and just set this.isDone = isDone. Then do null check will come at around line 80 like how they did with the other fields, inside the toModelType method

Reply:
the null check is to ensure which attribute is a compulsory field (This is based on the fact that null checks in model type throws missing field) . This exception is not for the json reading, but more for the other commands such as edit/add.